vassal-app/src/main/java/VASSAL/tools/io/DeobfuscatingInputStream.java

```diff
@@ -42,13 +44,13 @@ public DeobfuscatingInputStream(InputStream in) throws IOException {
     super(null);
 
     final byte[] header = new byte[ObfuscatingOutputStream.HEADER.length()];
-    readFully(in, header, 0, header.length);
+    readFully(in, header, HEADER_LENGTH);
```

Comment:
I wouldn't do it this way. The thing that `readFully` needs is the actual length of the array, which `header.length` will always be.